api/v1beta2/network_types.go

```diff
@@ -323,6 +323,11 @@ type VPCSpec struct {
 	// +kubebuilder:default=Ordered
 	// +kubebuilder:validation:Enum=Ordered;Random
 	AvailabilityZoneSelection *AZSelectionScheme `json:"availabilityZoneSelection,omitempty"`
+
+	// SecureDefaultVPCSecurityGroup specifies whether the default VPC security group ingress
+	// and egress rules should be removed.
+	// +optional
+	SecureDefaultVPCSecurityGroup bool `json:"secureDefaultVPCSecurityGroup,omitempty"`
```

Comment:
I'd remove this field on the next major release containing breaking changes. And change CAPA behavior as if this field was always true. Should we add a @todo somewhere?

Reply:
At least, we should add here in the description that the default will become true in the future, while currently it's false. Regarding a TODO, an issue could be opened for the next milestone.

Reply:
It might be nice if this is `true` for new clusters by default, but it stays `false` for older clusters. The defaulting can happen if in the validation webhook if the vpc id of a managed VPC is already filled.

Part of me was also considering of not adding this flag at all, which can be confusing?

Reply:
I'd prefer not adding the field/flag, because it makes things simpler and easier to understand. I don't think users should rely on this security group anyway. But someone in Slack suggested adding it to be backwards compatible, which also makes sense.